### ISSUE TITLE
Docs request: Diagrams for where whitespace rules can impact particular statements

### ISSUE THREAD

**Issue by platinumazure** [eslint/eslint#4564]
It could be pretty useful to have a diagram of various whitespace-related rules in the user guide and how they interact with each other. In addition, making this document could serve as a useful mental exercise for catching possible overlaps in these rules.

Examples might look something like the below, only prettier:

```js
  .--------------- space-after-keywords
  | .------------- space-in-parens
  | | .----------- space-infix-ops
  | | |   .------- space-infix-ops
  | | |   | .----- space-in-parens
  | | |   | | .--- space-before-blocks
  v v v   v v v
if ( x === 0 ) {
}
```

Basically the idea is, take some common statement types and show how it would be possible to configure their appearance by referencing related whitespace rules.

**Comment by eslintbot** [comment]
@platinumazure Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
This sounds nice in theory, in practice it's going to be a nightmare to maintain as rules added/changed/removed. We already have trouble keeping documentation in sync with rules, and I'm afraid adding a second spot to update when a subset of rules are changed will only lead to documentation that's always out-of-date.

**Comment by gyandeeps** [comment]
I agree with @nzakas

**Comment by mysticatea** [comment]
I like this idea, but I understand what @nzakas said.
How about below?

1. To create sub category about spacing rules in style rules.
   There are many spacing rules and other style rules, so this helps people looking up their favorite spacing setting.
2. To create spacing style setting page like the online demo. When someone changes setting, a sample code is modified by the `--fix` option, then it provides JSON (or YAML) setting.

**Comment by nzakas** [comment]
At one point, @ilyavolodin had an interactive demo that showed how spacing rules worked (I think).

**Comment by ilyavolodin** [comment]
I did. I never finished it. Although now with the `fix` options available, it might actually be pretty easy to do (although only for rules with autofixing).

**Comment by platinumazure** [comment]
@ilyavolodin Is it on GitHub? I would be happy to contribute if so.

**Comment by ilyavolodin** [comment]
@platinumazure Nope, never got it far enough to post it on github (and it was small too, not really worth it's own repository).

**Comment by gyandeeps** [comment]
@platinumazure @ilyavolodin Any one committing on doing this?
If not then we can close this.

**Comment by ilyavolodin** [comment]
Sorry, I can't commit to this right now. If @platinumazure wants to pick it up, I'd be perfectly fine with that, with the `fix` option it might be pretty easy to do a quick demo like that.

**Comment by nzakas** [comment]
Closing for now

**Comment by platinumazure** [comment]
Yeah, I won't have time to implement this unfortunately. Thanks for
discussing.
